src/cli/cli_coap.cpp

```diff
@@ -451,12 +728,31 @@ void Coap::HandleResponse(otMessage *aMessage, const otMessageInfo *aMessageInfo
         mInterpreter.mServer->OutputFormat("coap receive response error %d: %s\r\n", aError,
                                            otThreadErrorToString(aError));
     }
-    else
+    else if ((aMessageInfo != NULL) && (aMessage != NULL))
     {
-        mInterpreter.mServer->OutputFormat("coap response from ");
-        mInterpreter.OutputIp6Address(aMessageInfo->mPeerAddr);
+        otCoapOptionIterator iterator;
+
+        if (otCoapOptionIteratorInit(&iterator, aMessage) == OT_ERROR_NONE)
+        {
+            const otCoapOption *observeOpt =
```

Comment:
nit: add `OPENTHREAD_CONFIG_COAP_OBSERVE_API_ENABLE` around the observe-specific code here?

Reply:
More generally, it seems this PR adds a few hundred bytes of code when the observe feature is not enabled. Just looking for ways to minimize that overhead.

Reply:
Yep, understood, I found an extra spot where I was declaring and initialising an `otCoapOptionIterator`, but not using it because the offending code was `#if`'ed out by the `OPENTHREAD_CONFIG_COAP_OBSERVE_API_ENABLE` checks, so just widened those out a wee bit to capture the otherwise unused iterator as well.